py/objfloat.c

```diff
@@ -47,6 +47,13 @@
 #define M_PI (3.14159265358979323846)
 #endif
 
+// Workaround a bug in recent MSVC where NAN is no longer constant.
+// (By redefining back to the previous MSVC definition of NAN)
+#if defined(_MSC_VER) && _MSC_VER >= 1942
```

Comment:
It's tempting to make this `_MSC_VER == 1942` instead though on the other hand that might just lead to endless adjusting if the bug doesn't get fixed soon.

Reply:
I don't know much attention Microsoft pay to C rather than C++ (I think the new expression is constant in C++). So I wouldn't bet money it gets fixed quickly. :laughing: 

This should be harmless if it does stay in too long, AFAIK it's a valid definition of NAN and covered by tests that will pick up if it ever stops being one for some unexpected reason.